code/workspaces/web-app/src/components/app/TopBar.js

```diff
@@ -1,20 +1,65 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import AppBar from '@material-ui/core/AppBar';
-import Toolbar from '@material-ui/core/Toolbar';
+import { withStyles } from '@material-ui/core/styles';
 import UserInfo from './UserIcon';
+import datalabsLogo from '../../assets/images/datalabs-hori.png';
+import { extendSubdomain } from '../../core/getDomainInfo';
+import TopBarButton from './TopBarButton';
+import navBarLinks from '../../constants/navBarLinks';
 
-const TopBar = ({ identity }) => (
-  <AppBar position="fixed">
-    <Toolbar>
+const logoOrigWidth = 280;
+const logoOrigHeight = 50;
+const logoAspect = logoOrigWidth / logoOrigHeight;
+
+const styles = theme => ({
+  appBar: {
+    background: theme.palette.backgroundDark,
+    height: theme.shape.topBarHeight,
+    width: '100%',
+    display: 'block',
+    position: 'fixed',
+    top: 0,
+  },
+  toolBar: {
+    display: 'flex',
+    alignItems: 'center',
+    padding: `${theme.shape.topBarPaddingTopBottom}px ${theme.shape.topBarPaddingLeftRight}px`,
+    maxHeight: theme.shape.topBarContentHeight,
+  },
+  datalabsLogo: {
+    height: theme.shape.topBarContentHeight - theme.spacing(0.5),
+    width: (theme.shape.topBarContentHeight - theme.spacing(0.5)) * logoAspect,
```

Comment:
If you just set height and width as auto wouldn't the browser deal with the horizontal scaling rather than squash it?

Reply:
That was the way I tried to do it originally but for some reason the browser seemed to ignore the `auto` width and rendered it squashed. I'll take another look at it now I know that all the right classes are being applied in the right places to make sure that I didn't make a silly mistake with it!